README.md: clarify official tarballs vs. development

gperciva opened the pull request:
This uses stronger language (and even an emoji!) to direct people towards
using an official tarball, instead of trying to compile and use git master.

cperciva commented on README.md:
```diff
@@ -2,7 +2,13 @@ Client code for Tarsnap
+release tarball on https://www.tarsnap.com
+
+> This repository is intended for developers who may wish to watch changes in
+> progress or investigate bugs, not for production use.
```
I think we could add "or test new (unreleased) features" here?

gperciva commented:
Updated PR; I removed the "not for production use" since I think that's adequately implied by the other text, and keeping it in there would have been awkward ("... for developers who may wish to watch changes in progress, investigate bugs, or test new (unreleased) features, not for production use").
